pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1710,7 +1710,10 @@ private void refreshTopicPublishRate() {
     public void checkBrokerPublishThrottlingRate() {
         brokerPublishRateLimiter.checkPublishRate();
         if (brokerPublishRateLimiter.isPublishRateExceeded()) {
-            forEachTopic(topic -> ((AbstractTopic) topic).disableProducerRead());
+            forEachTopic(topic -> {
+                topic.increasePublishLimitedTimes();
```

Comment:
This is duplicated count. Line 1715 will trigger `ServerCnx#disableCnxAutoRead`.